components/validator/validator.go

```diff
@@ -129,23 +130,23 @@ func (v *Validator) Stop() {
 func (v *Validator) loop() {
 	defer v.wg.Done()
 
-	ticker := time.NewTicker(v.cfg.PollInterval)
-	defer ticker.Stop()
+	if !v.cfg.OutputSubmitterDisabled {
+		trySubmitOutputCh := make(chan struct{})
+		go func() {
+			for {
+				trySubmitOutputCh <- struct{}{}
+			}
+		}()
 
-	for {
-		select {
-		case <-ticker.C:
-			if !v.cfg.OutputSubmitterDisabled {
+		for {
+			select {
+			case <-trySubmitOutputCh:
 				if err := v.submitL2Output(); err != nil {
 					v.l.Error("failed to submit l2 output", "err", err)
 				}
+			case <-v.ctx.Done():
+				return
 			}
-
-			if err := v.submitChallengeTx(); err != nil {
-				v.l.Error("failed to submit challenge tx", "err", err)
-			}
```

Comment:
Submitting challenge is about to implemented in another PR outside of this `for` statement, so I remove it.